File: day1/elevator.rb
directions = File.readlines('./input', :chomp=>true)[0]

floor = 0
floor += directions.count(?() - directions.count(?))

puts "floor #{floor}"

floor = 0
directions.chars.each_index do |i|
  case directions[i]
  when '('
    floor += 1
  when ')'
    floor -= 1
  end
  if floor < 0
    puts "position #{i + 1}"
    break
  end
end

File: day1/elevator.py
if __name__ == '__main__':

    with open('input') as f:
        inst = f.read()
        print(inst.count('(') - inst.count(')'))

    floor = 0
    for i in range(len(inst)):
        if inst[i] == '(':
            floor += 1
        elif inst[i] == ')':
            floor -= 1
            if floor < 0:
                print(i+1)
                break   
        else:
            print(f"unexpected char, found {inst[i]}")